debugdrawer/src/main/java/io/palaima/debugdrawer/util/DebugDrawerLifecycleCallbacks.java

```diff
@@ -0,0 +1,62 @@
+package io.palaima.debugdrawer.util;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import io.palaima.debugdrawer.DebugDrawer;
+
+public class DebugDrawerLifecycleCallbacks implements Application.ActivityLifecycleCallbacks {
+    private Activity mActivity;
+    private DebugDrawer mDebugDrawer;
+
+    public DebugDrawerLifecycleCallbacks(Activity activity, DebugDrawer debugDrawer) {
+        mActivity = activity;
+        mDebugDrawer = debugDrawer;
+    }
+
+    @Override
+    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
+
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+        if (mActivity == activity) {
+            mDebugDrawer.onStart();
+        }
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        if (mActivity == activity) {
+            mDebugDrawer.onResume();
+        }
+    }
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        if (mActivity == activity) {
```

Comment:
Since the lifecycle callbacks and the debug drawer are installed for that particular instance of an Activity I think you would have to check if they are actually the same instance. So object equality is needed here